src/configuration.cpp

```diff
@@ -451,20 +453,36 @@ chimera::CompiledConfiguration::CompiledConfiguration(
     // Override individual templates if specified in the configuration.
     if (bindingNode_)
     {
-        if (const YAML::Node classTemplateNode = bindingNode_["class"])
-            bindingDefinition_.class_cpp = Lookup(classTemplateNode);
+        if (const auto node = lookupYAMLNode(bindingNode_, "class", "header"))
+            bindingDefinition_.class_h = Lookup(node);
+
+        if (const auto node = lookupYAMLNode(bindingNode_, "class", "source"))
+            bindingDefinition_.class_cpp = Lookup(node);
+
+        if (const auto node = lookupYAMLNode(bindingNode_, "enum", "header"))
+            bindingDefinition_.enum_h = Lookup(node);
+
+        if (const auto node = lookupYAMLNode(bindingNode_, "enum", "source"))
+            bindingDefinition_.enum_cpp = Lookup(node);
 
-        if (const YAML::Node enumTemplateNode = bindingNode_["enum"])
-            bindingDefinition_.enum_cpp = Lookup(enumTemplateNode);
+        if (const auto node
+            = lookupYAMLNode(bindingNode_, "function", "header"))
+            bindingDefinition_.function_h = Lookup(node);
 
-        if (const YAML::Node functionTemplateNode = bindingNode_["function"])
-            bindingDefinition_.function_cpp = Lookup(functionTemplateNode);
+        if (const auto node
+            = lookupYAMLNode(bindingNode_, "function", "source"))
+            bindingDefinition_.function_cpp = Lookup(node);
 
-        if (const YAML::Node moduleTemplateNode = bindingNode_["module"])
-            bindingDefinition_.module_cpp = Lookup(moduleTemplateNode);
+        if (const auto node = lookupYAMLNode(bindingNode_, "module"))
+            bindingDefinition_.module_cpp = Lookup(node);
```

Comment:
Should we allow header and source on module just for consistency?  It is a bit abnormal but maybe it's ok?

Reply:
Sounds fair to me. Done in https://github.com/personalrobotics/chimera/pull/289/commits/468e6aa490e4aa1292e01e876cd052a4e009bc2f.